ts/src/error.ts

```diff
@@ -15,16 +15,28 @@ export class ProgramError extends Error {
     err: any,
     idlErrors: Map<number, string>
   ): ProgramError | null {
+    const errString: string = err.toString();
     // TODO: don't rely on the error string. web3.js should preserve the error
     //       code information instead of giving us an untyped string.
-    let components = err.toString().split("custom program error: ");
-    if (components.length !== 2) {
-      return null;
+    let unparsedErrorCode: string;
+    if (errString.includes("custom program error:")) {
+      let components = errString.split("custom program error: ");
+      if (components.length !== 2) {
+        return null;
+      } else {
+        unparsedErrorCode = components[1];
+      }
+    } else {
+      const matches = errString.match(/"Custom":([0-9]+)}/g);
```

Comment:
When will we hit this branch and not the first branch?

Reply:
with real tx instead of simulated ones

Reply:
thanks